MANIFEST.in

```diff
@@ -0,0 +1,12 @@
+include *.py
+include *.rc
+include *.rst
+include *.txt
+include *.yapf
+include .isort.cfg
+include mypy.ini
+include tox.ini
+recursive-include docs *.py
+recursive-include docs *.rst
+recursive-include docs Makefile
+recursive-include tests *.py
```

Comment:
Sorry, I noticed this only here. Why do you want to include tests in the distribution? Would this mean that the tests are also shipped to pypi.org?

Reply:
I personally put my tests inside the actual package and distribute them everywhere the code goes.  But sure, lots of people don't.  I'll look into this because perhaps the tests are getting installed even when you just `pip install .`.

Reply:
https://github.com/Parquery/pylddwrap/blob/60b8b05073081e050a1a6758e8993e83241a994c/setup.py#L46

Ok, so you do exclude the tests from installation.  Do you feel it is bad to include the tests in the sdist?  It seems fairly reasonable to me to have them available for people that get grabbing the source.

To directly respond to your questions, it is useful to have tests available wherever you have the code so that no matter how you have the code you can run the tests.  Sure, most people won't leverage this.  They pay the price of 36K or whatever of disk space used.  A bare venv with nothing extra installed is 15MB so... Again, sure, lots of packages don't do this.  I don't know where I would even go for an authoritative should or shouldn't on this but the age old distutils says it defaults to including them.

https://docs.python.org/3/distutils/sourcedist.html#specifying-the-files-to-distribute
> anything that looks like a test script: test/test*.py (currently, the Distutils don’t do anything with test scripts except include them in source distributions, but in the future there will be a standard for testing Python module distributions)

As to them being shipped to PyPI.  Yes, if they are either in the sdist or the wheel then the expectation would be that they end up on PyPI.